### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by martinvonk**
# Short Description

- Section on Pastas organisation with reference to PastaStore and Metran
- Reference to STOWA manual
- Reference to related Python packages
- Remove numbered prefix so we don't have to rename (all) files after a change.

# Checklist before PR can be merged:
- [x] closes (part of) issue #545

### CODE REVIEW

File: doc/about/courses.rst [hunk 1]
```diff
@@ -1,10 +1,9 @@
-On-site-courses
+Courses
 ===============
```
Comment (raoulcollenteur) [inline doc/about/courses.rst:2]: underline should have same length as title, change throughout.

File: doc/about/users.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Users
```
Comment (raoulcollenteur) [inline doc/about/users.rst:1]: Let's remove this file and add this later. It should be internationally oriented as well.

**Review by raoulcollenteur — CHANGES_REQUESTED** [review]

**Comment by raoulcollenteur** [comment]
Some nice additions, but also some new errors .. ;-)

Not a big fan of all the manual linking to rst files and notebooks. Why is this better? This is prone to errors, multiple links are already not working now because of this:

https://pastas--560.org.readthedocs.build/en/560/examples/index.html

A possible fix would be to implement nblink or some url checker. Also, there are new warnings given by ReadTheDocs that should be fixed first.

File: doc/about/users.rst (hunk 2 above)
Comment (martinvonk) [reply doc/about/users.rst:1]: There should be international drinking water and engineering companies right? :)

But we should discuss first whether or not we want to add a list of users before adding it so I'll remove it for now.

**Comment by martinvonk** [comment]
> Not a big fan of all the manual linking to rst files and notebooks. Why is this better? 

Now if you remove notebook `01_basic_model.ipynb` from the examples folder with files:
```
00_prepare_timeseries.ipynb,
01_basic_model.ipynb,
02_fix_parameters.ipynb,
03_calibration_options.ipynb,
04_adding_rivers.ipynb,
05_adding_wells.ipynb,
06_multiple_wells.ipynb,
07_adding_trends.ipynb,
08_changing_responses.ipynb,
09_threshold_non_linear.ipynb,
10_non_linear_recharge.ipynb,
11_recharge_estimation.ipynb,
12_snowmodel.ipynb,
13_comparing_models.ipynb,
14_diagnostic_checking.ipynb,
15_timestep_analysis.ipynb,
16_uncertainty.ipynb,
17_emcee_uncertainty.ipynb,
18_sgi_example.ipynb,
19_signatures.ipynb
```
you have to rename:

```
00_prepare_timeseries.ipynb -> 00_prepare_timeseries.ipynb
02_fix_parameters.ipynb -> 01_fix_parameters.ipynb
03_calibration_options.ipynb -> 02_calibration_options.ipynb
04_adding_rivers.ipynb -> 03_adding_rivers.ipynb
05_adding_wells.ipynb -> 04_adding_wells.ipynb
06_multiple_wells.ipynb -> 05_multiple_wells.ipynb
07_adding_trends.ipynb -> 06_adding_trends.ipynb
08_changing_responses.ipynb -> 07_changing_responses.ipynb
09_threshold_non_linear.ipynb -> 08_threshold_non_linear.ipynb
10_non_linear_recharge.ipynb -> 09_non_linear_recharge.ipynb
11_recharge_estimation.ipynb -> 10_recharge_estimation.ipynb
12_snowmodel.ipynb -> 11_snowmodel.ipynb
13_comparing_models.ipynb -> 12_comparing_models.ipynb
14_diagnostic_checking.ipynb -> 13_diagnostic_checking.ipynb
15_timestep_analysis.ipynb -> 14_timestep_analysis.ipynb
16_uncertainty.ipynb -> 15_uncertainty.ipynb
17_emcee_uncertainty.ipynb -> 16_emcee_uncertainty.ipynb
18_sgi_example.ipynb -> 17_sgi_example.ipynb
19_signatures.ipynb -> 18_signatures.ipynb
```
With this change, you can simply adjust the `index.rst` file in the examples folder

**Comment by martinvonk** [comment]
> Also, there are new warnings given by ReadTheDocs that should be fixed first.

I'll have a look at these.

**Comment by raoulcollenteur** [comment]
> > Not a big fan of all the manual linking to rst files and notebooks. Why is this better?
> 
> Now if you remove notebook `01_basic_model.ipynb` from the examples folder with files:
> 
> ```
> 00_prepare_timeseries.ipynb,
> 01_basic_model.ipynb,
> 02_fix_parameters.ipynb,
> 03_calibration_options.ipynb,
> 04_adding_rivers.ipynb,
> 05_adding_wells.ipynb,
> 06_multiple_wells.ipynb,
> 07_adding_trends.ipynb,
> 08_changing_responses.ipynb,
> 09_threshold_non_linear.ipynb,
> 10_non_linear_recharge.ipynb,
> 11_recharge_estimation.ipynb,
> 12_snowmodel.ipynb,
> 13_comparing_models.ipynb,
> 14_diagnostic_checking.ipynb,
> 15_timestep_analysis.ipynb,
> 16_uncertainty.ipynb,
> 17_emcee_uncertainty.ipynb,
> 18_sgi_example.ipynb,
> 19_signatures.ipynb
> ```
> 
> you have to rename:
> 
> ```
> 00_prepare_timeseries.ipynb -> 00_prepare_timeseries.ipynb
> 02_fix_parameters.ipynb -> 01_fix_parameters.ipynb
> 03_calibration_options.ipynb -> 02_calibration_options.ipynb
> 04_adding_rivers.ipynb -> 03_adding_rivers.ipynb
> 05_adding_wells.ipynb -> 04_adding_wells.ipynb
> 06_multiple_wells.ipynb -> 05_multiple_wells.ipynb
> 07_adding_trends.ipynb -> 06_adding_trends.ipynb
> 08_changing_responses.ipynb -> 07_changing_responses.ipynb
> 09_threshold_non_linear.ipynb -> 08_threshold_non_linear.ipynb
> 10_non_linear_recharge.ipynb -> 09_non_linear_recharge.ipynb
> 11_recharge_estimation.ipynb -> 10_recharge_estimation.ipynb
> 12_snowmodel.ipynb -> 11_snowmodel.ipynb
> 13_comparing_models.ipynb -> 12_comparing_models.ipynb
> 14_diagnostic_checking.ipynb -> 13_diagnostic_checking.ipynb
> 15_timestep_analysis.ipynb -> 14_timestep_analysis.ipynb
> 16_uncertainty.ipynb -> 15_uncertainty.ipynb
> 17_emcee_uncertainty.ipynb -> 16_emcee_uncertainty.ipynb
> 18_sgi_example.ipynb -> 17_sgi_example.ipynb
> 19_signatures.ipynb -> 18_signatures.ipynb
> ```
> 
> With this change, you can simply adjust the `index.rst` file in the examples folder because al the files are listed there. Yes, you have to be tidy with the bookkeeping but that should not be too big of an issue.

I hope you wrote a script to write that comment ;-) I can see your point.

**Comment by raoulcollenteur** [comment]
checking consistency... /home/docs/checkouts/readthedocs.org/user_builds/pastas/checkouts/560/doc/examples/diagnostic_checking.ipynb: WARNING: document isn't included in any toctree

**Comment by raoulcollenteur** [comment]
Perhaps ReadTheDocs should throwing errors when these kind of things come up somehow. Now we have to manually check for errors in the built process to uncover this.

**Comment by martinvonk** [comment]
> I hope you wrote a script to write that comment ;-) I can see your point.

👼 .
Also, not renaming the files constantly helps with the GitHub commit history.

**Comment by martinvonk** [comment]
> Perhaps ReadTheDocs should throwing errors when these kind of things come up somehow. Now we have to manually check for errors in the built process to uncover this.

Agree

**Comment by martinvonk** [comment]
> checking consistency... /home/docs/checkouts/readthedocs.org/user_builds/pastas/checkouts/560/doc/examples/diagnostic_checking.ipynb: WARNING: document isn't included in any toctree

Fixed this, should be good now.

**Comment by raoulcollenteur** [comment]
Nice work. Merging.

**Review by raoulcollenteur — APPROVED** [review]

**Pull request merged**